File: LensInfo.py
import DevAddr as DA
import DefVal as DV
import LensCtrl

def GeneralInfo():
    print("\n***** General Information *****")

    retval, version = LensCtrl.ProtocolVersion()
    if (retval == DV.RET_SUCCESS):
        print("Protocol version = ", version)
    else:
        print(retval)
    
    retval, version = LensCtrl.FWVersion()
    if (retval == DV.RET_SUCCESS):
        print("Firmware version = ", version)
    else:
        print(retval)
    
    retval, revision = LensCtrl.LensRevision()
    if (retval == DV.RET_SUCCESS):
        print("Lens revision = ", revision)
    else:
        print(retval)

    retval, i2cAddr = LensCtrl.LensAddress()
    if (retval == DV.RET_SUCCESS):
        print("Lens I2C address = {:#04x} (7bit)" .format(i2cAddr))
    else:
        print(retval)

    retval, model = LensCtrl.ModelName()
    if (retval == DV.RET_SUCCESS):
        print("Lens model name = ", model)
    else:
        print(retval)
    
    retval, capabilities = LensCtrl.CapabilitiesRead()
    if (retval == DV.RET_SUCCESS):
        print("Capabilities = {:#06x} : {:#016b}" .format(capabilities, capabilities))
    else:
        print(retval)

    retval, status1 = LensCtrl.Status1Read()
    if (retval == DV.RET_SUCCESS):
        print("Status1      = {:#06x} : {:#016b}" .format(status1, status1))
    else:
        print(retval)
    
    retval, status2 = LensCtrl.Status2ReadSet()
    if (retval == DV.RET_SUCCESS):
        print("Status2      = {:#06x} : {:#016b}" .format(status2, status2))
    else:
        print(retval)

    retval, kelvinValue = LensCtrl.TempKelvinVal()
    if (retval == DV.RET_SUCCESS):
        tempCels = round(kelvinValue - DV.KELVIN_TO_CELS)
        print("Temperature = ", tempCels, "degree Celsius")
    else:
        print(retval)

def ZoomInfo():
    print("\n***** ZOOM Information *****")
    print("Current postion = ", LensCtrl.zoomCurrentAddr)
    retval, flag = LensCtrl.ZoomBacklashRead()
    print("Backlash corrention = ", flag)
    pos = LensCtrl.NoErrChk2BytesIntRead(DA.ZOOM_INIT_POSITION)
    print("Initialize position = ", pos)
    min = LensCtrl.NoErrChk2BytesIntRead(DA.ZOOM_MECH_STEP_MIN)
    print("Mechanical min = ", min)
    max = LensCtrl.NoErrChk2BytesIntRead(DA.ZOOM_MECH_STEP_MAX)
    print("Mechanical max = ", max)
    print("Position min = ", LensCtrl.zoomMinAddr)
    print("Position max = ", LensCtrl.zoomMaxAddr)
    print("Speed value = ", LensCtrl.zoomSpeedPPS)
    retval, speedMin = LensCtrl.ZoomSpeedMinRead()
    print("Speed min = ", speedMin)
    retval, speedMax = LensCtrl.ZoomSpeedMaxRead()
    print("Speed max = ", speedMax)
    retval, count = LensCtrl.ZoomCountValRead()
    print("Count value = ", count)
    retval, countMax = LensCtrl.ZoomCountMaxRead()
    print("Count max = ", countMax)



def FocusInfo():
    print("\n***** FOCUS Information *****")
    print("Current position = ", LensCtrl.focusCurrentAddr)
    retval, flag = LensCtrl.FocusBacklashRead()
    print("Backlash correction = ", flag)
    pos = LensCtrl.NoErrChk2BytesIntRead(DA.FOCUS_INIT_POSITION)
    print("Initialize position = ", pos)
    print("Mechanical min = ", LensCtrl.focusMinAddr)
    print("Mechanical max = ", LensCtrl.focusMaxAddr)
    min = LensCtrl.NoErrChk2BytesIntRead(DA.FOCUS_POSITION_MIN)
    print("Position min = ", min)
    max = LensCtrl.NoErrChk2BytesIntRead(DA.FOCUS_POSITION_MAX)
    print("Position max = ", max)
    print("Speed value = ", LensCtrl.focusSpeedPPS)
    retval, speedMin = LensCtrl.FocusSpeedMinRead()
    print("Speed min = ", speedMin)
    retval, speedMax = LensCtrl.FocusSpeedMaxRead()
    print("Speed max = ", speedMax)
    retval, count = LensCtrl.FocusCountValRead()
    print("Count value = ", count)
    retval, countMax = LensCtrl.FocusCountMaxRead()
    print("Count max = ", countMax)

def IrisInfo():
    print("\n***** IRIS Information *****")
    print("Current position = ", LensCtrl.irisCurrentAddr)
    retval, flag = LensCtrl.IrisBacklashRead()
    print("Backlash correction = ", flag)
    pos = LensCtrl.NoErrChk2BytesIntRead(DA.IRIS_INIT_POSITION)
    print("Initialize position = ", pos)
    print("Mechanical min = ", LensCtrl.irisMinAddr)
    print("Mechanical max = ", LensCtrl.irisMaxAddr)
    min = LensCtrl.NoErrChk2BytesIntRead(DA.IRIS_POSITION_MIN)
    print("Position min = ", min)
    max = LensCtrl.NoErrChk2BytesIntRead(DA.IRIS_POSITION_MAX)
    print("Position max = ", max)
    print("Speed value = ", LensCtrl.irisSpeedPPS)
    retval, speedMin = LensCtrl.IrisSpeedMinRead()
    print("Speed min = ", speedMin)
    retval, speedMax = LensCtrl.IrisSpeedMaxRead()
    print("Speed max = ", speedMax)
    retval, count = LensCtrl.IrisCountValRead()
    print("Count value = ", count)
    retval, countMax = LensCtrl.IrisCountMaxRead()
    print("Count max = ", countMax)

def OptFilterInfo():
    print("\n***** OPTICAL FILTER Information *****")
    print("Current position = ", LensCtrl.optCurrentAddr)
    pos = LensCtrl.NoErrChk2BytesIntRead(DA.OPT_FILTER_INIT_POSITION)
    print("Initialize position = ", pos)
    print("Mechanical min = ", 0)
    print("Mechanical max = ", LensCtrl.optFilMaxAddr)
    print("Position min = ", 0)
    max = LensCtrl.NoErrChk2BytesIntRead(DA.OPT_FILTER_POSITION_MAX)
    print("Position max = ", max)
    retval, count = LensCtrl.OPtFilterCountValRead()
    print("Count value = ", count)
    retval, countMax = LensCtrl.OptFilterCountMaxRead()
    print("Count max = ", countMax)

File: LensConnect_SampleProgram_Linux_C_x86_x64_1.0.0/LensConnect_Controller/C/LensInfo.c
 #include<limits.h>
#include"LensCtrl.h"
#include"LensConnect.h"

void PrintBinaly(uint16_t value) {
	int i = 0;
	unsigned int mask = (int)1 << (sizeof(value) * CHAR_BIT - 1);
	do {
		putchar(mask & value ? '1' : '0');
		i++;
		if (i == 4){
			putchar(' ');
			i = 0;
		}
	} while (mask >>= 1);
	printf("\n");
}

void GeneralInfo() {
	int retval;
	char version[7];
	char model[25];

	printf("\n***** General Information *****\n");

	retval = ProtocolVersion(version);
	if (retval == SUCCESS) {
		printf("Protocol version  = %s\n", version);
	}
	else {
		printf("%s", ErrorTxt(retval));
	}

	retval = FWVersion(version);
	if (retval == SUCCESS) {
		printf("Firemware version = %s\n", version);
	}
	else {
		printf("%s", ErrorTxt(retval));
	}
	
	int revision = 0;
	retval = LensRevision(&revision);
	if (retval == SUCCESS) {
		printf("Lens revision = %d\n", revision);
	}
	else {
		printf("%s", ErrorTxt(retval));
	}
	
	int i2cAddr = 0;
	retval = LensAddress(&i2cAddr);
	if (retval == SUCCESS) {
		printf("Lens I2C address = 0x%x (7bit)\n", i2cAddr);
	}
	else {
		printf("%s", ErrorTxt(retval));
	}
	
	retval = ModelName(model);
	if (retval == SUCCESS) {
		printf("Lens model name = %s\n", model);
	}
	else {
		printf("%s", ErrorTxt(retval));
	}
	uint16_t capabilities;
	retval = CapabilitiesRead(&capabilities);
	if (retval == SUCCESS) {
		printf("Capabilities = ");
		PrintBinaly(capabilities);
	}
	else {
		printf("%s", ErrorTxt(retval));
	}
	
	uint16_t status1 = 0;
	retval = Status1Read(&status1);
	if (retval == SUCCESS) {
		printf("Status1 = ");
		PrintBinaly(status1);
	}
	else {
		printf("%s", ErrorTxt(retval));
	}
	
	retval = Status2ReadSet();
	if (retval == SUCCESS) {
		printf("status2 = "); 
		PrintBinaly(status2);
	}
	else {
		printf("%s", ErrorTxt(retval));
	}
	
	int kelvinValue = 0;
	retval = TempKelvin(&kelvinValue);
	if (retval == SUCCESS) {
		int tempCels = (int)round(kelvinValue - KELVIN_TO_CELS);
		printf("Temperature = %u degree Celsius\n", tempCels);
	}
	else {
		printf("%s", ErrorTxt(retval));
	}
}
void ZoomInfo() {
	uint16_t flag;
	uint16_t speedPPS;
	uint32_t count;
	printf("\n***** ZOOM Information *****\n");
	printf("Current position = %d\n", zoomCurrentAddr);
	ZoomBacklashRead(&flag);
	printf("Backlash correction = %s\n", OnOff(flag));
	printf("Initialize position = %d\n", NoErrChk2BytesRead(ZOOM_INIT_POSITION));
	printf("Mechanical min = %d\n", NoErrChk2BytesRead(ZOOM_MECH_STEP_MIN));
	printf("Mechanical max = %d\n", NoErrChk2BytesRead(ZOOM_MECH_STEP_MAX));
	printf("Position min = %d\n", zoomMinAddr);
	printf("Position max = %d\n", zoomMaxAddr);
	printf("Speed value = %d\n", zoomSpeedPPS);
	ZoomSpeedMinRead(&speedPPS);
	printf("Speed min   = %d\n", speedPPS);
	ZoomSpeedMaxRead(&speedPPS);
	printf("Speed max   = %d\n", speedPPS);
	ZoomCountValRead(&count);
	printf("Count value = %d\n", count);
	ZoomCountMaxRead(&count);
	printf("Count max   = %d\n", count);

}
void FocusInfo() {
	uint16_t flag;
	uint16_t speedPPS;
	uint32_t count;
	printf("\n***** FOCUS Information *****\n");
	printf("Current position = %d\n", focusCurrentAddr);
	FocusBacklashRead(&flag);
	printf("Backlash correction = %s\n", OnOff(flag));
	printf("Initialize position = %d\n", NoErrChk2BytesRead(FOCUS_INIT_POSITION));
	printf("Mechanical min = %d\n", focusMinAddr);
	printf("Mechanical max = %d\n", focusMaxAddr);
	printf("Position min = %d\n", NoErrChk2BytesRead(FOCUS_POSITION_MIN));
	printf("Position max = %d\n", NoErrChk2BytesRead(FOCUS_POSITION_MAX));
	printf("Speed value = %d\n", focusSpeedPPS);
	FocusSpeedMinRead(&speedPPS);
	printf("Speed min   = %d\n", speedPPS);
	FocusSpeedMaxRead(&speedPPS);
	printf("Speed max   = %d\n", speedPPS);
	FocusCountValRead(&count);
	printf("Count value = %d\n", count);
	FocusCountMaxRead(&count);
	printf("Count max   = %d\n", count);
}
void IrisInfo() {
	uint16_t flag;
	uint16_t speedPPS;
	uint32_t count;
	printf("\n***** IRIS Information *****\n");
	printf("Current position = %d\n", irisCurrentAddr);
	IrisBacklashRead(&flag);
	printf("Backlash correction = %s\n", OnOff(flag));
	printf("Initialize position = %d\n", NoErrChk2BytesRead(IRIS_INIT_POSITION));
	printf("Mechanical min = %d\n", irisMinAddr);
	printf("Mechanical max = %d\n", irisMaxAddr);
	printf("Position min = %d\n", NoErrChk2BytesRead(IRIS_POSITION_MIN));
	printf("Position max = %d\n", NoErrChk2BytesRead(IRIS_POSITION_MAX));
	printf("Speed value = %d\n", irisSpeedPPS);
	IrisSpeedMinRead(&speedPPS);
	printf("Speed min   = %d\n", speedPPS);
	IrisSpeedMaxRead(&speedPPS);
	printf("Speed max   = %d\n", speedPPS);
	IrisCountValRead(&count);
	printf("Count value = %u\n", count);
	IrisCountMaxRead(&count);
	printf("Count max   = %u\n", count);
}
void OptFilterInfo() {
	uint32_t count;
	printf("\n***** OPTICAL FILTER Information *****\n");
	printf("Current position = %d\n", optCurrentAddr);
	printf("Initialize position = %d\n", NoErrChk2BytesRead(OPT_FILTER_INIT_POSITION));
	printf("Mechanical min = %d\n", 0);
	printf("Mechanical max = %d\n",optFilMaxAddr);
	printf("Position min = %d\n", 0);
	printf("Position max = %d\n", NoErrChk2BytesRead(OPT_FILTER_POSITION_MAX));
	OptFilterCountValRead(&count);
	printf("Count value = %d\n", count);
	OptFilterCountMaxRead(&count);
	printf("Count max   = %d\n", count);
}
